lib/widgets/actions.dart

```diff
@@ -31,6 +31,29 @@ abstract final class ZulipAction {
   /// for details on the UI feedback, see there.
   static Future<void> markNarrowAsRead(BuildContext context, Narrow narrow) async {
     final zulipLocalizations = ZulipLocalizations.of(context);
+    final store = PerAccountStoreWidget.of(context);
+
+    final unreadCount = switch (narrow) {
+      CombinedFeedNarrow() => store.unreads.countInCombinedFeedNarrow(),
+      ChannelNarrow(:final streamId) => store.unreads.countInChannelNarrow(streamId),
+      MentionsNarrow() => store.unreads.countInMentionsNarrow(),
+      KeywordSearchNarrow() => store.unreads.countInKeywordSearchNarrow(),
+      StarredMessagesNarrow() => store.unreads.countInStarredMessagesNarrow(),
+      _ => 0,
+    };
+
+    if (narrow is CombinedFeedNarrow || narrow is ChannelNarrow ||
+      narrow is MentionsNarrow || narrow is KeywordSearchNarrow ||
+      narrow is StarredMessagesNarrow) {
+      final didConfirm = showSuggestedActionDialog(context: context,
```

Comment:
Please update the dartdoc of `ZulipAction.markNarrowAsRead` so that it mentions this new thing it does.